Orange/data/domain.py

```diff
@@ -58,69 +69,101 @@ def __init__(self, source, destination):
 
 
 class Domain:
-    def __init__(self, attributes, class_vars=None, metas=None, source=None):
+    def __init__(self, attributes=(), class_vars=(), metas=(), source=None):
```

Comment:
The beauty here is that one can construct superset domains by specifying only some `attributes` as `str`, for example, while leaving the others as `Variable`. Thus only the strings are sourced.

A separate `from_domain()` method would more strictly imply a domain subset, I guess, but I sure am not against it.